### ISSUE TITLE
Add flag to skip generation of `x-go-name`

### ISSUE THREAD

**Issue by aeneasr** [go-swagger/go-swagger#2119]
## Problem statement

It would be great to skip generation of fields `x-go-name` and `x-go-package` as they cause problematic behavior when two types from different packages have the same name.

## Environment
swagger version: 0c1371f870a2c1b90025387d25f160b4f1b50a5a
go version: 1.13

**Comment by casualjim** [comment]
I think it might be good to add that to the codescanner, there are a few places where code like this appears: https://github.com/go-swagger/go-swagger/blob/master/codescan/schema.go#L211-L216

adding a flag to disable that would probably solve the issues you're experiencing

**Comment by aeneasr** [comment]
Yeah, that makes sense. I've written a small sanitizer which simply removes those fields from the generated spec file, which is good enough for me for the moment :)

**Comment by casualjim** [comment]
Just for other people looking:

```sh
sed -n '/x-go-name/!p' ./swagger.json
sed -n '/x-go-package/!p' ./swagger.json 
```
should do the trick

**Comment by aeneasr** [comment]
I think that might cause issues where `x-go-(name|package)` is the last element (trailing comma from entry before).

**Comment by casualjim** [comment]
You're right, with jq:

```sh
cat swagger.json | jq 'walk(if type == "object" then del(.["x-go-package","x-go-name"]) else . end)' > /tmp/rewrite.json && mv /tmp.rewrite.json swagger.json
```

**Comment by sashayakovtseva** [comment]
This also leaks internal details one may not want to share.

**Comment by shearn89** [comment]
Just bumped into this issue - I'm generating my spec based off annotations, and when I try to load it into AWS API Gateway it complains that x-go-package is an unsupported keyword. I know they could fix it at their end too, but having the option to skip it would be nice!

**Comment by nevins-b** [comment]
Looks like setting `SWAGGER_GENERATE_EXTENSION=false` in the environment stops these from being generated https://github.com/go-swagger/go-swagger/blob/56218e58d0952f191d71705d25dfe0e6ca0d9d3b/codescan/schema.go#L21

**Comment by pcgeng** [comment]
> Looks like setting `SWAGGER_GENERATE_EXTENSION=false` in the environment stops these from being generated
> 
> https://github.com/go-swagger/go-swagger/blob/56218e58d0952f191d71705d25dfe0e6ca0d9d3b/codescan/schema.go#L21

How to enable it on Windows? Need to config it in environment variables?

**Comment by pcgeng** [comment]
> > Looks like setting `SWAGGER_GENERATE_EXTENSION=false` in the environment stops these from being generated
> > https://github.com/go-swagger/go-swagger/blob/56218e58d0952f191d71705d25dfe0e6ca0d9d3b/codescan/schema.go#L21
> 
> How to enable it on Windows? Need to config it in environment variables?

`set SWAGGER_GENERATE_EXTENSION=false`
